src/time/TimeContext.tsx

```diff
@@ -0,0 +1,39 @@
+import {useInterval} from '@atb/utils/use-interval';
+import {clock} from '@entur-private/abt-time-react-native-lib';
+import React, {createContext, useContext, useState} from 'react';
+
+type TimeContextState = {
+  /**
+   * The current time in milliseconds, updated every 2.5 seconds. This is based
+   * on server time when possible, and can be safely used for checking the
+   * validity of fare contracts.
+   */
+  now: number;
+};
+
+const TimeContext = createContext<TimeContextState | undefined>(undefined);
+
+export const TimeContextProvider: React.FC = ({children}) => {
+  const [now, setNow] = useState(Date.now());
+  useInterval(() => clock.currentTimeMillis().then(setNow), 2500);
```

Comment:
Should probably send the `triggerImmediately` argument as `true` to `useInterval` to avoid 2,5s where the time comes from `Date.now()`

Reply:
Also, what if the device is offline?
Perhaps the `TimeContextProvider` should separate between something like `serverNow` and `deviceNow`.
Supporting both seem to be a good idea imo, and `serverNow` could then be undefined as default?

Reply:
> what if the device is offline?

The library isn't very transparent about this right now, and it seems like it also fails silently by returning device time if it can't fetch from the server. So since we can't trust the clock 100% anyway, I think having the default to Date.now is easier to use in the app, since we never have to add fallback logic like `serverTime || deviceTime`.

Reply:
Oh right! Good to catch that :+1:
Regarding `triggerImmediately`, maybe the default value in the hook should be true? Perhaps it was unintuitive to even consider this when using it?

Reply:
Looks like it's close to 50/50 true/false on use of `triggerImmediately` in the app. Although it can be easy to forget,  I like that default is the same as javascripts setInterval now.